service/common/src/main/java/org/apache/polaris/service/catalog/io/FileIOFactory.java

```diff
@@ -18,10 +18,50 @@
  */
 package org.apache.polaris.service.catalog.io;
 
+import jakarta.annotation.Nonnull;
+import jakarta.enterprise.context.ApplicationScoped;
 import java.util.Map;
+import java.util.Set;
+import org.apache.iceberg.catalog.TableIdentifier;
 import org.apache.iceberg.io.FileIO;
+import org.apache.polaris.core.context.RealmId;
+import org.apache.polaris.core.persistence.PolarisResolvedPathWrapper;
+import org.apache.polaris.core.storage.PolarisStorageActions;
 
-/** Interface for providing a way to construct FileIO objects, such as for reading/writing S3. */
+/**
+ * Interface for providing a way to construct FileIO objects, such as for reading/writing S3.
+ *
+ * <p>Implementations are available via CDI as {@link ApplicationScoped @ApplicationScoped} beans.
+ */
 public interface FileIOFactory {
-  FileIO loadFileIO(String impl, Map<String, String> properties);
+
+  /** This method is intended for use in tests only. */
```

Comment:
... but it is called from `BasePolarisCatalog`. I think we ought to refactor that call path to avoid having extension point APIs that have to provide different functionality in test contexts (can be done later).

Reply:
Done!